tests/test_orientation.py

```diff
@@ -0,0 +1,82 @@
+"""Tests for marker orientations."""
+
+import math
+from pathlib import Path
+
+import pytest
+from pytest import approx
+
+from sb_vision import FileCamera, Orientation, Vision
+
+ROTATION_TOLERANCE_DEGREES = 6
+
+CALIBRATIONS = Path(__file__).parent.parent / 'calibrations' / 'tecknet_rotations'
+
+TEST_IMAGES = [
+    (
+        '0rot_x-45rot_y1.1z.png',
+        Orientation(rot_x=0, rot_y=math.radians(-45), rot_z=0),
+    ),
+    (
+        '-45rot_x0rot_y1.1z.png',
+        Orientation(rot_x=math.radians(-45), rot_y=0, rot_z=0),
+    ),
+    (
+        '-22.5rot_x0rot_y0.6z.png',
+        Orientation(rot_x=math.radians(-22.5), rot_y=0, rot_z=0),
+    ),
+    (
+        '0rot_x-22.5rot_y0.6z.png',
+        Orientation(rot_x=0, rot_y=math.radians(-22.5), rot_z=0),
+    ),
+    (
+        '0rot_x0rot_y0.55z.png',
+        Orientation(rot_x=0, rot_y=0, rot_z=0),
+    ),
+    (
+        '-90rot_z.png',
+        Orientation(rot_x=0, rot_y=0, rot_z=math.radians(270)),
+    ),
+    (
+        '90rot_z.png',
+        Orientation(rot_x=0, rot_y=0, rot_z=math.radians(90)),
+    ),
+    (
+        '180rot_z.png',
+        Orientation(rot_x=0, rot_y=0, rot_z=math.radians(180)),
+    ),
+    (
+        '135rot_z.png',
+        Orientation(rot_x=0, rot_y=0, rot_z=math.radians(135)),
+    ),
+    (
+        '45rot_z.png',
+        Orientation(rot_x=0, rot_y=0, rot_z=math.radians(45)),
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    "photo, expected_orientation",
+    TEST_IMAGES,
+)
+def test_image_coordinates(photo, expected_orientation):
+    """Make sure that this particular file gives these particular tokens."""
+    camera = FileCamera(CALIBRATIONS / photo, camera_model='C016')
+    vision = Vision(camera)
+    token, = vision.snapshot()
+
+    def approx_ang(expected):
+        return approx(expected, abs=ROTATION_TOLERANCE_DEGREES)
+
+    def assert_angle(angle, expected, message):
+        expected = approx_ang(math.degrees(expected))
+        # Check both +0 and +360 so approx can cover the jump between -180 and 180
+        assert math.degrees(angle) == expected or math.degrees(angle) + 360 == expected, \
```

Comment:
I'd much rather we were explicit about whether we expect `-90` or `270`. I don't mind which we go with, but I think the tests should validate one or the other rather than both.

Reply:
Fixed in 361ce0935cc83e803f00956705613c8b85a0a387